langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -486,25 +500,28 @@ function testCloneWithTypeJsonRec1() {
     json|error ss = p.cloneWithType(json);
     assert(ss is json, true);
 
-    json j = <json> ss;
-    assert(j.toJsonString(), "{\"name\":\"N\", \"age\":3}");
+    if (ss is json) {
+        assert(ss.toJsonString(), "{\"name\":\"N\", \"age\":3}");
+    }
 }
 
 function testCloneWithTypeJsonRec2() {
    json pj = { name : "tom", age: 2};
    Person2|error pe = pj.cloneWithType(Person2);
    assert(pe is Person2, true);
 
-   Person2 p = <Person2> pe;
-   assert(p.name, "tom");
-   assert(p.age, 2);
+   if (pe is Person2) {
```

Comment:
Aren't we silently ignoring the error here? I think we should either do 

```
Person2 p = checkpanic pe;
```

or add an else block and fail there.

Reply:
`assert(pe is Person2, true);` is there, therefore do we need to do the changes suggested above?

Reply:
Sorry, my bad, this is OK. But instead of using the type test shall we use `checkpanic` directly?

Reply:
Changed